packages/commonwealth/client/scripts/state/api/threads/helpers/cache.ts

```diff
@@ -231,6 +231,26 @@ const cacheUpdater = ({
   method,
   arrayManipulationMode = 'replaceArray',
 }: CacheUpdater) => {
+  // TODO: research a simpler cache invalidation strategy for tRPC queries
+  void queryClient.invalidateQueries({
+    predicate: (query) => {
+      const [path, args] = query.queryKey;
+      if (Array.isArray(path) && path.length === 2) {
+        const [entity, name] = path;
+        if (entity === 'thread' && name === 'getThreadsByIds') {
+          const { input } = args as {
+            input: { community_id: string; thread_ids: string };
+          };
+          return (
+            input.community_id === communityId &&
+            input.thread_ids?.split(',').includes(threadId.toString())
+          );
+        }
+      }
+      return false;
+    },
+  });
```

Comment:
1. again, this kind of TODO left in the code is not actionable without ticket.
2. @mzparacha could you take a look on this piece?  There is a lot of description at the beginning of this file, we should validate if this is still true. Also, I am not sure if we shuold invalidate queries here as this function look like manual cache updater. Having said that, I agree that this file is super complicated, but this was because the complexity we had around thread models and different ways of fetch threads from api. Maybe something has changed after @Rotorsoft changes and we can simplify it?

Reply:
Same as above

Reply:
2. 
> There is a lot of description at the beginning of this file, we should validate if this is still true.

Some of it is not true as of today, many of the extra thread APIs have now been removed. Those comments need to be updated.

I also agree that we shouldn't invalidate the whole cache for 1 object change. Invalidation seems better DX, but worse UX. But again its not simple to update so I am fine if we quickly followup with a plan to correctly update the cache.